vault/resource_generic_secret.go

```diff
@@ -104,14 +118,50 @@ func genericSecretResourceWrite(d *schema.ResourceData, meta interface{}) error
 	}
 
 	log.Printf("[DEBUG] Writing generic Vault secret to %s", path)
-	_, err = client.Logical().Write(path, data)
+	secret, err := client.Logical().Write(path, data)
 	if err != nil {
 		return fmt.Errorf("error writing to Vault: %s", err)
 	}
 
 	d.SetId(path)
 
-	return genericSecretResourceRead(d, meta)
+	captureResponse := d.Get("capture_response").(bool)
+	if captureResponse {
+		log.Printf("[DEBUG] Capture response is set.")
+		if secret == nil {
+			log.Printf("[WARN] write to (%s) returned empty response", path)
+			return nil
+		}
+		jsonDataBytes, err := json.Marshal(secret.Data)
+		log.Printf("[DEBUG] Secret is %s", jsonDataBytes)
+		if err != nil {
+			return fmt.Errorf("Error marshaling JSON for %q: %s", path, err)
+		}
+		d.Set("data_json", string(jsonDataBytes))
+
+		// Since our "data" map can only contain string values, we
+		// will take strings from Data and write them in as-is,
+		// and write everything else in as a JSON serialization of
+		// whatever value we get so that complex types can be
+		// passed around and processed elsewhere if desired.
+		dataMap := map[string]string{}
+		for k, v := range secret.Data {
+			if vs, ok := v.(string); ok {
+				dataMap[k] = vs
+			} else {
```

Comment:
You can get rid of the `else` block if you put a `continue` in the `if` block.

```
if vs, ok := v.(string); ok {
	dataMap[k] = vs
        continue
}
// Again ignoring error because we know this value
// came from JSON in the first place and so must be valid.
vBytes, _ := json.Marshal(v)
dataMap[k] = string(vBytes)
```